BackEndLib/Wchar.cpp

```diff
@@ -885,3 +893,45 @@ WSTRING WCSReplace(
 		processed = loc + from.size();
 	}
 }
+
+//*****************************************************************************
+WSTRING WCSToLower(WSTRING const &source)
+//Converts WSTRING to lowercase
+{
+	WSTRING lowercased;
+	lowercased.reserve(source.size());
+	for (WSTRING::const_iterator it = source.begin(); it != source.end(); ++it)
+		lowercased += towlower(*it);
+
+	return lowercased;
```

Comment:
C++11 range-based `for` is cleaner here:

```suggestion
	for (char c : source) 
		lowercased += towlower(c);

	return lowercased;
```